### PR TITLE
[Function optimization] add docstring of five eager method

### PR DESCRIPTION
**Pull request by wj-Mcat**
### PR types

Function optimization

### PR changes

Docs

### Describe

add docstring of three eager method: 

- numpy
- clear_gradient
- detach
- clone
- element_size

### CODE REVIEW

**Comment by SigureMo** [comment]
我今天在本地编译了一下发现是可行的～

image

UPDATE:

raw literal string（`R"DOC` 这种）多行字符串的形式也是可以的，但是需要注意不能多换行（比如 `numpy()` 不能放到第二行），不能多空格（比如 `--` 前面不能有空格），否则就会因为解析不出签名而认为没有签名

image

**Comment by wj-Mcat** [comment]
你上面的这个结论是你测试出来的吧？有没有参考依据呢？

我在这边找到了python c api中的相关文档，你可以尝试一下：

https://docs.python.org/3.7/c-api/intro.html#c.PyDoc_STRVAR

https://peps.python.org/pep-0007/


基于 PyDoc_STRVAR来实现，或许可以更稳定，且能够做一些重构。

**Comment by SigureMo** [comment]
> 基于 PyDoc_STRVAR来实现，或许可以更稳定，且能够做一些重构。

根据 PEP7 中写的宏定义，这是没有必要的吧，

> ```cpp
> #ifndef PyDoc_STR
> #define PyDoc_VAR(name)         static char name[]
> #define PyDoc_STR(str)          (str)
> #define PyDoc_STRVAR(name, str) PyDoc_VAR(name) = PyDoc_STR(str)
> #endif
> ```

PEP7 是一个 CPython 的 C 代码风格指南（主要是要求 CPython 内部开发规范指南），对于外部是没有强制性的，而且根据其宏定义，是没有额外操作的，只是规范代码风格而已，~~在当前用例可能显得有些冗余~~

UPDATE:

多写几个后发现直接写在注册那里可能显得有点乱 :joy:，所以现在把 Docstring 对应方法放在了一起

**Comment by SigureMo** [comment]
目前 `paddle/fluid/pybind/imperative.cc`（VarBase） 所拥有的 5 个方法的 Docstring 已经都放到 `paddle/fluid/pybind/eager_method.cc`（EagerTensor） 了，均可通过 `method.__doc__` 访问 Docstring，均可通过 `inspect.signature(method)` 获取签名，均可显示在文档里（预览链接 <http://preview-paddle-pr-49630.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/Tensor_en.html>）

这里调整了下文档中不合适和错误的地方，主要包含：

- 替换示例代码中不必要的 NumPy API 使用；
- `clear_gradient` 加上参数 `set_to_zero`（仅位置参数），这个参数中文文档也是没有的，只有代码里能看出来，中文那边也需要修正下；
- 格式微调。

UPDATE:

给 Tensor 类本身也加上了 `__doc__`，使得英文文档和中文文档对齐。

**Comment by wj-Mcat** [comment]
这边统计了下eager_method.cc 代码中开放函数的支持情况：

|     method               |     文档情况          |
|--------------------------|-----------------------|
|     numpy                |     ✅                 |
|     copy_                |     没有相关文档      |
|     clone                |     ✅                 |
|     reconstruct_from_    |     没找到相关文档    |
|     retain_grads         |     没找到相关文档    |
|     clear_gradient       |     ✅                 |
|     is_dense             |     没找到相关文档    |
|     detach               |     ✅                 |
|     get_tensoar          |     没有相关文档      |
|     get_selected_rows    |     没有相关文档      |
|     set_string_list      |     没有相关文档      |
|     set_vocab            |     没有相关文档      |
|     get_map_tensor       |     没有相关文档      |
|     nnz                  |     没有相关文档      |
|     indices              |     没有相关文档      |
|     values               |     没有相关文档      |
|     crows                |     没有相关文档      |
|     cols                 |     没有相关文档      |
|     is_sparse            |     没有相关文档      |
|     is_sparse_coo        |     没有相关文档      |
|     is_sparse_csr        |     没有相关文档      |
|     is_same_shape        |     没有相关文档      |
|     to_sparse_csr        |     没有相关文档      |
|     element_size         |     ✅                 |
|     is_selected_rows     |     没有相关文档      |
|     rows                 |     没有相关文档      |
|     data_ptr             |     没有相关文档      |

相关工作也已经完成，辛苦 @SigureMo ，麻烦 @wanghuancoder 帮忙看看，非常感谢 

此外，没有相关文档的函数后续是需要支持还是？？？ @wanghuancoder

File: paddle/fluid/pybind/eager_method.cc [hunk 1]
```diff
@@ -284,6 +284,27 @@ static PyObject* tensor_method_numpy(TensorObject* self,
...
+
+        import paddle
+
+        data = paddle.uniform([30, 10, 32], dtype="float32", min=-1, max=1)
+        linear = paddle.nn.Linear(32, 64)
+        data = paddle.to_tensor(data)
+        x = linear(data)
+        print(x.numpy())
+)DOC");
+
```
Comment (wanghuancoder) [inline paddle/fluid/pybind/eager_method.cc]: 我觉得doc的声明，放在函数体之前比较合适。这样与Python端文档在前，代码在后的顺序一致~

Comment (SigureMo) [reply paddle/fluid/pybind/eager_method.cc]: > 我觉得doc的声明，放在函数体之前比较合适。这样与Python端文档在前，代码在后的顺序一致~

同意～我稍后改一下～

**Comment by luotao1** [comment]
是否可以将这部分内容更新到 RFC 文档里

**Comment by SigureMo** [comment]
> 是否可以将这部分内容更新到 RFC 文档里

嗯嗯，这部分之后会一起更新进去
